pkg/dbrole/dbrole.go

```diff
@@ -77,13 +110,22 @@ func indexOf(dbRole DbRole) int {
 	return -1
 }
 
+type DbRoleSlice []DbRole // Needed for sorting records
 // Returns true if the first role has fewer permissions than the second
 // role, and true if the two roles are the same or the second role has
 // more permissions.
-func (a DbRoleSlice) Less(i, j int) bool {
-	return indexOf(a[i]) <= indexOf(a[j])
+func (a DbRoleSlice) Less(i, j int) bool { return indexOf(a[i]) <= indexOf(a[j]) }
+
+func (a DbRoleSlice) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
+
+func (a DbRoleSlice) Len() int { return len(a) }
+
+func Max(dbRoles []DbRole) DbRole {
+	sort.Sort(DbRoleSlice(dbRoles))
+	return dbRoles[len(dbRoles)-1]
 }
 
-func (a DbRoleSlice) Swap(i, j int) {
-	a[i], a[j] = a[j], a[i]
+func Min(dbRoles []DbRole) DbRole {
+	sort.Sort(DbRoleSlice(dbRoles))
+	return dbRoles[0]
```

Comment:
`Max` & `Min` functions currently would have a side effect (sort the input slice). There is really no reason to do that; these functions should be very simple. I propose you create a copy of the given slice and then sort that slice so that the input slice is not modified.

Reply:
will fix it

Reply:
Noted, can we take it up later as we don't use metadata based roles?